nni/tools/nnictl/launcher.py

```diff
@@ -509,6 +509,11 @@ def launch_experiment(args, experiment_config, mode, experiment_id):
     rest_process, start_time = start_rest_server(args.port, experiment_config['trainingServicePlatform'], \
                                                  mode, experiment_id, foreground, log_dir, log_level)
     nni_config.set_config('restServerPid', rest_process.pid)
+    # save experiment information
+    nnictl_experiment_config = Experiments()
+    nnictl_experiment_config.add_experiment(experiment_id, args.port, start_time,
+                                            experiment_config['trainingServicePlatform'],
+                                            experiment_config['experimentName'], pid=rest_process.pid, logDir=log_dir)
```

Comment:
why put this code snippet here? what type of error will be induced if keep it in the previous place?

Reply:
due to now we get `port` and `pid` from `.experiment`, we need set `pid` in `.experiment` when setting `pid` in `.config`.

Reply:
got it